resources/packs/aws/aws_cloudfront.go

```diff
@@ -0,0 +1,129 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go-v2/service/cloudfront"
+	"github.com/cockroachdb/errors"
+	"go.mondoo.com/cnquery/resources/packs/core"
+)
+
+func (f *mqlAwsCloudfront) id() (string, error) {
+	return "aws.cloudfront", nil
+}
+
+func (f *mqlAwsCloudfrontDistribution) id() (string, error) {
+	return f.Arn()
+}
+
+func (f *mqlAwsCloudfront) GetDistributions() ([]interface{}, error) {
+	provider, err := awsProvider(f.MotorRuntime.Motor.Provider)
+	if err != nil {
+		return nil, err
+	}
+	// global service
+	svc := provider.Cloudfront("")
+	ctx := context.Background()
+	res := []interface{}{}
+
+	var marker *string
+	for {
+		distributions, err := svc.ListDistributions(ctx, &cloudfront.ListDistributionsInput{Marker: marker})
+		if err != nil {
+			return nil, errors.Wrap(err, "could not gather aws cloudfront distributions")
+		}
+
+		for i := range distributions.DistributionList.Items {
+			d := distributions.DistributionList.Items[i]
+			origins, err := core.JsonToDict(d.Origins)
```

Comment:
In GCP for such dicts I used manual mapping of the API object to JSON. If we just convert the API object directly to dict, we risk breaking the MQL resource when updating the AWS SDK library. Perhaps the question is do we treat the dict fields as stable MQL resources or not?

Reply:
that's a good point. i'll switch this up

Reply:
i ended up converting some of the former dict fields into fields on the resource or a list of defined resources. left the cache behaviors as a straight object -> dict translation, the caching info for the resource